src/bag_exporter.cpp

```diff
@@ -146,24 +153,30 @@ void BagExporter::export_bag()
   }
 
   // Get topic metadata
-  auto metadata = reader.get_all_topics_and_types();
+  auto metadata = reader.get_metadata().topics_with_message_count;
 
   // Initialize handlers based on available topics
+  size_t total_bag_messages = 0;
   for (auto & [topic_name, handler] : handlers_) {
     auto it = std::find_if(metadata.begin(), metadata.end(),
-      [&topic_name](const rosbag2_storage::TopicMetadata & tm) {
-        return tm.name == topic_name;
+      [&topic_name](const rosbag2_storage::TopicInformation & ti) {
+        return ti.topic_metadata.name == topic_name;
       });
     if (it == metadata.end()) {
-      RCLCPP_WARN(this->get_logger(), "Topic '%s' not found in the bag.", topic_name.c_str());
+      RCLCPP_WARN(this->get_logger(), "Topic '%s' not found in the bag \U0000274C", topic_name.c_str());
```

Comment:
Probably too late now (And I risk the wrath of emojis here), but I would put all emoji codes in a helper/util library and use their names for easier modification and introspection (Wtf is `\U0000274C`?).
Feel free to ignore/resolve this comment.